youtube_dl/utils.py

```diff
@@ -1270,7 +1270,10 @@ def date_from_str(date_str):
         unit += 's'
         delta = datetime.timedelta(**{unit: time})
         return today + delta
-    return datetime.datetime.strptime(date_str, '%Y%m%d').date()
+    try:
+        return datetime.datetime.strptime(date_str, '%Y%m%d').date()
+    except ValueError:
+        sys.exit('ERROR: cannot parse datestring ' + date_str)
```

Comment:
Never ever do that. You must handle it at the place of actual call the method.

Reply:
Could you please briefly explain, why this exception should be handled in the surrounding of every function call of date_from_string() instead of inside the function?
Catching the exception inside the function should produce less code.

Reply:
Cause it's an **utility function** and it's used in other places you've broken with this change.